lib/timeutils/wallclocktime.c

```diff
@@ -1029,3 +1039,274 @@ dump_wall_clock_time(const WallClockTime *wct, GString *output)
+    case 'S':
+      val = wct->wct_sec;
+      goto number;
+    case 't':
+      *l = 1;
+      return "\t";
+    case 'T':
+      fmt = "%H:%M:%S";
+      goto recu_strftime;
+    case 'u':
+      val = wct->wct_wday ? wct->wct_wday : 7;
+      width = 1;
+      goto number;
+    case 'U':
+      val = (wct->wct_yday + 7U - wct->wct_wday) / 7;
+      goto number;
+    case 'W':
+      val = (wct->wct_yday + 7U - (wct->wct_wday+6U)%7) / 7;
+      goto number;
+    case 'V':
+      val = wall_clock_time_iso_week_number(wct);
+      goto number;
+    case 'w':
+      val = wct->wct_wday;
+      width = 1;
+      goto number;
+    case 'x':
+      fmt = _TIME_LOCALE(loc)->d_fmt;
+      goto nl_strftime;
+    case 'X':
+      fmt = _TIME_LOCALE(loc)->t_fmt;
+      goto nl_strftime;
+    case 'y':
+      val = (wct->wct_year + 1900LL) % 100;
+      if (val < 0) val = -val;
+      goto number;
+    case 'Y':
+      val = wct->wct_year + 1900LL;
+      if (val >= 10000)
+        {
+          *l = snprintf(*s, sizeof *s, "+%lld", val);
+          return *s;
+        }
+      width = 4;
+      goto number;
+    case 'z':
+      if (wct->wct_gmtoff == -1)
+        {
+          *l = 0;
+          return "";
+        }
+      *l = snprintf(*s, sizeof *s, "%+.4ld",
+                    wct->wct_gmtoff/3600*100 + wct->wct_gmtoff%3600/60);
+      return *s;
+    case 'Z':
+      if (wct->wct_gmtoff == -1)
+        {
+          *l = 0;
+          return "";
+        }
+      *l = snprintf(*s, sizeof *s, "%c%02ld:%02ld",
+                    wct->wct_gmtoff < 0 ? '-' : '+',
+                    wct->wct_gmtoff/3600, wct->wct_gmtoff%3600/60);
+      return *s;
+    case '%':
+      *l = 1;
+      return "%";
+    default:
+      return 0;
+    }
+number:
+  switch (pad ? pad : def_pad)
+    {
+    case '-':
+      *l = snprintf(*s, sizeof *s, "%lld", val);
+      break;
+    case '_':
+      *l = snprintf(*s, sizeof *s, "%*lld", width, val);
+      break;
+    case '0':
+    default:
+      *l = snprintf(*s, sizeof *s, "%0*lld", width, val);
+      break;
+    }
+  return *s;
+nl_strcat:
+string:
+  *l = strlen(fmt);
+  return fmt;
+nl_strftime:
+recu_strftime:
+  *l = wall_clock_time_strftime(wct, *s, sizeof *s, fmt);
+  if (!*l) return 0;
+  return *s;
+}
+
+size_t
+wall_clock_time_strftime(WallClockTime *wct, char *s, size_t n, const char *f)
+{
+  size_t l, k;
+  char buf[100];
+  char *p;
+  const char *t;
+  int pad, plus;
+  long width;
+  for (l=0; l<n; f++)
+    {
+      if (!*f)
+        {
+          s[l] = 0;
+          return l;
+        }
+      if (*f != '%')
+        {
+          s[l++] = *f;
+          continue;
+        }
+      f++;
+      pad = 0;
+      if (*f == '-' || *f == '_' || *f == '0') pad = *f++;
+      if ((plus = (*f == '+'))) f++;
+      width = (int) strtoul(f, &p, 10);
+      if (*p == 'C' || *p == 'F' || *p == 'G' || *p == 'Y')
+        {
+          if (!width && p!=f) width = 1;
+        }
+      else if (*p == 'f')
```

Comment:
Can we add the musl implementation without modifying it (marking their commit id in our commit message), and apply the modifications in a later patch so that we can review and follow upstream changes easier?